### PR TITLE
✨ (scatter) don't animate as a time scatter

### PR DESCRIPTION
**Pull request by sophiamersmann**
Animates a scatter as a cloud of moving dots and keeps the axes fixed. The axes are fixed for the particular time range that is animated. 

At the animation's beginning and end, the view "jumps" to the new axis domains. The same happens when an animation is paused and the user starts dragging one of the sliders.

The animation in relative mode is unchanged. Getting the right axis domain doesn't seem trivial to me unless I'm missing something, so it's not worth doing at this point IMO.

In a follow-up PR, I'll also look into fixing the size domain.

### CODE REVIEW

**Comment by sophiamersmann** [comment]
* **#4038** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/4038?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sophiamersmann and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by marcelgerber — APPROVED** [review]
Well done, I like it!

File: packages/@ourworldindata/grapher/src/scatterCharts/ScatterPlotChart.tsx [hunk 1]
```diff
@@ -299,6 +298,57 @@ export class ScatterPlotChart
...
+
+        // get y-values and exclude non-positive values if on log scale
+        let yValues = table.get(this.yColumnSlug).uniqValues
+        if (this.yScaleType === ScaleType.log) {
+            yValues = yValues.filter((value) => value > 0)
+        }
+
+        return {
+            x: domainExtent(xValues, this.xScaleType),
+            y: domainExtent(yValues, this.yScaleType),
```
[the remark this answers is not in the document]
Comment (sophiamersmann) [reply packages/@ourworldindata/grapher/src/scatterCharts/ScatterPlotChart.tsx:348]: Sweet!

File: packages/@ourworldindata/grapher/src/scatterCharts/ScatterPlotChart.tsx [hunk 2]
```diff
@@ -299,6 +298,57 @@ export class ScatterPlotChart
...
+        )
+
+        if (
+            addCountryMode === EntitySelectionMode.Disabled ||
+            addCountryMode === EntitySelectionMode.SingleEntity
+        ) {
+            table = table.filterByEntityNames(
+                this.selectionArray.selectedEntityNames
+            )
+        }
```
Comment (marcelgerber) [inline packages/@ourworldindata/grapher/src/scatterCharts/ScatterPlotChart.tsx:328]: This can go now, after #4055.

**Comment by sophiamersmann** [comment]
### Merge activity

* **Oct 22, 8:56 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/4038).

**Pull request merged**